go/vt/vttablet/tabletserver/query_executor.go

```diff
@@ -958,7 +960,7 @@ func (qre *QueryExecutor) execRevertMigration() (*sqltypes.Result, error) {
 	return qre.tsv.onlineDDLExecutor.SubmitMigration(qre.ctx, qre.plan.FullStmt)
 }
 
-func (qre *QueryExecutor) execShowMigrations() (*sqltypes.Result, error) {
+func (qre *QueryExecutor) execShowMigrations(conn *StatefulConnection) (*sqltypes.Result, error) {
```

Comment:
@shlomi-noach is `conn *StatefulConnection` meant to be used here? I don't follow this part

Reply:
Yes. I'm not all too familiar with this either, but I got guidance from @harshit-gangal. The `StatefulConnection` is the standard way for you to implement a transactional behavior working with QueryService. So we should pass that variable. Normally I'd also _use_ the variable -- except:
- I don't really require `show vitess_migrations` to exhibit transactional behavior; it's a `show` command and I don;t need it to e.g. be repeatable-read or anything. I only need it to work from within a transaction.
- Therefore no need to bloat the transaction.
- Also, the existing logic uses the executor's own connection pooling, which is fine, and there's no need/reason to now take apart the existing and working mechanism within the executor to accommodate the transactional connection.
-

Reply:
Ok sounds good 👍